### PR TITLE
[TS migration] Migrate HTMLEngineProvider component

### PR DESCRIPTION
**Pull request by ruben-rebelo**
### Details
[TS migration] Migrate HTMLEngineProvider component

### Fixed Issues

$ https://github.com/Expensify/App/issues/25097

PROPOSAL: N/A

### Tests
- [x] Verify that no errors appear in the JS console

**How to test**

1. Open a chat
2. Send/View the following types of messages:
- Email address
- Website
- Attachment
- Image
- Mention of user
- here tag mention
- Task with checkbox

All of the above should work as normally, they work if you press them depending on the functionality

### Offline tests

N/A

### QA Steps
- [x] Verify that no errors appear in the JS console

Same as in the **Tests** section.

### PR Author Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/39693995/e90362ea-2d14-4ab7-88f2-8b94c8faf3ee

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/39693995/4cba5258-99f5-4ecd-9588-7b9b8dabc859

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/39693995/16907bb6-4942-45cf-b348-ba2115787ec6

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/39693995/ce6faf2a-6008-47fb-96fb-52a60bd30cc1

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/39693995/17e82e9b-4d89-4d2f-8231-ee741e1e67bf

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/39693995/ece68ab0-4e95-41d2-9ef4-0e2ef93e9194

</details>

### CODE REVIEW

File: src/components/HTMLEngineProvider/index.native.tsx [hunk 1]
```diff
@@ -1,20 +1,18 @@
 import React from 'react';
 import BaseHTMLEngineProvider from './BaseHTMLEngineProvider';
-import {defaultProps, propTypes} from './htmlEnginePropTypes';
+import {HTMLEngineProviderProps} from './types';
 
-function HTMLEngineProvider(props) {
+function HTMLEngineProvider({debug = false, children = null}: HTMLEngineProviderProps) {
```
Comment (VickyStash) [inline src/components/HTMLEngineProvider/index.native.tsx:5]:
```suggestion
function HTMLEngineProvider({debug = false, children}: HTMLEngineProviderProps) {
```

Comment (VickyStash) [inline src/components/HTMLEngineProvider/index.native.tsx:5]:
```suggestion
function HTMLEngineProvider({debug = false, children}: HTMLEngineProviderProps) {
```

File: src/components/HTMLEngineProvider/BaseHTMLEngineProvider.tsx [hunk 2]
```diff
@@ -1,88 +1,90 @@
...
+                contentModel: HTMLContentModel.textual,
             }),
-            'mention-user': defaultHTMLElementModels.span.extend({tagName: 'mention-user'}),
-            'mention-here': defaultHTMLElementModels.span.extend({tagName: 'mention-here'}),
-            'next-step': defaultHTMLElementModels.span.extend({
-                tagName: 'next-step',
+            'mention-user': HTMLElementModel.fromCustomModel({tagName: 'mention-user', contentModel: HTMLContentModel.textual}),
+            'mention-here': HTMLElementModel.fromCustomModel({tagName: 'mention-here', contentModel: HTMLContentModel.textual}),
+            'next-steps': HTMLElementModel.fromCustomModel({
+                tagName: 'next-steps',
```
Comment (VickyStash) [inline src/components/HTMLEngineProvider/BaseHTMLEngineProvider.tsx:63]: Why `step` turned into `steps`?

File: src/components/HTMLEngineProvider/BaseHTMLEngineProvider.tsx [hunk 3]
```diff
@@ -1,88 +1,90 @@
...
                 mixedUAStyles: {...styles.textLabelSupporting},
+                contentModel: HTMLContentModel.textual,
             }),
-            'next-step-email': defaultHTMLElementModels.span.extend({tagName: 'next-step-email'}),
-            video: defaultHTMLElementModels.div.extend({
+            'next-steps-email': HTMLElementModel.fromCustomModel({tagName: 'next-steps-email', contentModel: HTMLContentModel.textual}),
```
Comment (VickyStash) [inline src/components/HTMLEngineProvider/BaseHTMLEngineProvider.tsx:67]: Same, why `step` turned into `steps`?

File: src/components/HTMLEngineProvider/BaseHTMLEngineProvider.tsx [hunk 4]
```diff
@@ -1,88 +1,90 @@
...
-function BaseHTMLEngineProvider(props) {
+function BaseHTMLEngineProvider({textSelectable = false, children, enableExperimentalBRCollapsing = false}: BaseHTMLEngineProviderProps) {
     const styles = useThemeStyles();
 
     // Declare nonstandard tags and their content model here
+    /* eslint-disable @typescript-eslint/naming-convention */
     const customHTMLElementModels = useMemo(
         () => ({
-            edited: defaultHTMLElementModels.span.extend({
+            edited: HTMLElementModel.fromCustomModel({
```
Comment (VickyStash) [inline src/components/HTMLEngineProvider/BaseHTMLEngineProvider.tsx:31]: @ruben-rebelo Please:
1. Explain why these updates are necessary
2. Test it extensively to make sure it works the same way as before.

File: src/components/HTMLEngineProvider/BaseHTMLEngineProvider.tsx (hunk 2 above)
Comment (ruben-rebelo) [reply src/components/HTMLEngineProvider/BaseHTMLEngineProvider.tsx:63]: This is a Typo.

File: src/components/HTMLEngineProvider/BaseHTMLEngineProvider.tsx (hunk 3 above)
Comment (ruben-rebelo) [reply src/components/HTMLEngineProvider/BaseHTMLEngineProvider.tsx:67]: Same here, this is a Typo.

File: src/components/HTMLEngineProvider/BaseHTMLEngineProvider.tsx (hunk 4 above)
Comment (ruben-rebelo) [reply src/components/HTMLEngineProvider/BaseHTMLEngineProvider.tsx:31]: 1. This was changed because the extend function do not allow the change of the tagName, in order to change the tagName of the component we must generate a new element based on the contentModel.
2. I'll test it a bit more since next-steps were wrongly created, but so far all the elements seem to be correctly changed on the database and displayed on the platforms. Will let you know when I'm sure nothing will break.

Comment (ruben-rebelo) [reply src/components/HTMLEngineProvider/BaseHTMLEngineProvider.tsx:31]: @VickyStash Regarding the testing, I can confirm that the elements are displayed correctly and store correctly on Onyx database.

**Review by VickyStash — APPROVED** [review]
LGTM

File: src/components/HTMLEngineProvider/BaseHTMLEngineProvider.tsx [hunk 5]
```diff
@@ -1,88 +1,90 @@
...
 // Configuration for RenderHTML is handled in a top-level component providing
 // context to RenderHTMLSource components. See https://git.io/JRcZb
 // Beware that each prop should be referentialy stable between renders to avoid
 // costly invalidations and commits.
-function BaseHTMLEngineProvider(props) {
+function BaseHTMLEngineProvider({textSelectable = false, children, enableExperimentalBRCollapsing = false}: BaseHTMLEngineProviderProps) {
     const styles = useThemeStyles();
 
     // Declare nonstandard tags and their content model here
+    /* eslint-disable @typescript-eslint/naming-convention */
```
Comment (blazejkustra) [inline src/components/HTMLEngineProvider/BaseHTMLEngineProvider.tsx:28]: Why is this needed?

**Review by blazejkustra — APPROVED** [review]

File: src/components/HTMLEngineProvider/BaseHTMLEngineProvider.tsx (hunk 5 above)
Comment (VickyStash) [reply src/components/HTMLEngineProvider/BaseHTMLEngineProvider.tsx:28]: Otherwise eslint gives error for `customHTMLElementModels` [params](https://github.com/ruben-rebelo/App/blob/ts-migration/HTMLEngineProvider-component/src/components/HTMLEngineProvider/BaseHTMLEngineProvider.tsx#L35-L44) names like `alert-text`, `muted-text`, etc

File: src/components/HTMLEngineProvider/htmlEngineUtils.ts [hunk 6]
```diff
@@ -7,12 +8,12 @@ const MAX_IMG_DIMENSIONS = 512;
...
- * @param {string} tagName - The name of the tag for which max width should be constrained.
- * @param {number} contentWidth - The content width provided to the HTML
+ * @param tagName - The name of the tag for which max width should be constrained.
+ * @param contentWidth - The content width provided to the HTML
  * component.
- * @returns {number} The minimum between contentWidth and MAX_IMG_DIMENSIONS
+ * @returns The minimum between contentWidth and MAX_IMG_DIMENSIONS
  */
-function computeEmbeddedMaxWidth(tagName, contentWidth) {
+function computeEmbeddedMaxWidth(contentWidth: number, tagName: string): number {
```
Comment (s77rt) [inline src/components/HTMLEngineProvider/htmlEngineUtils.ts:16]: This is interesting were we doing something wrong the whole time? Isn't this param swapping a breaking change?

File: src/components/HTMLEngineProvider/htmlEngineUtils.ts [hunk 7]
```diff
@@ -7,12 +8,12 @@ const MAX_IMG_DIMENSIONS = 512;
  * is used by the HTML component in the default renderer for img tags to scale
  * down images that would otherwise overflow horizontally.
  *
- * @param {string} tagName - The name of the tag for which max width should be constrained.
- * @param {number} contentWidth - The content width provided to the HTML
+ * @param tagName - The name of the tag for which max width should be constrained.
+ * @param contentWidth - The content width provided to the HTML
```
Comment (s77rt) [inline src/components/HTMLEngineProvider/htmlEngineUtils.ts:12]: NAB. Swap the comments lines, first param goes first

**Review by s77rt — CHANGES_REQUESTED** [review]

**Comment by ruben-rebelo** [comment]
Hello @s77rt, Thank you so much for your review, I've agreed and applied your suggestions, please take a look.

File: src/components/HTMLEngineProvider/types.ts [hunk 8]
```diff
@@ -1,13 +1,7 @@
...
+import type ChildrenProps from '@src/types/utils/ChildrenProps';
 
-type HTMLEngineProviderProps = {
-    /** Children to wrap in HTMLEngineProvider. */
-    children?: ReactNode;
-
-    /** Optional debug flag. Prints the TRT in the console when true. */
-    debug?: boolean;
-};
+type HTMLEngineProviderProps = ChildrenProps;
```
Comment (s77rt) [inline src/components/HTMLEngineProvider/types.ts:4]: We don't need to define a type that is just another type. We can just use `ChildrenProps` and remove `HTMLEngineProviderProps`

File: src/components/HTMLEngineProvider/index.tsx [hunk 9]
```diff
@@ -2,19 +2,12 @@ import React from 'react';
 import useWindowDimensions from '@hooks/useWindowDimensions';
 import * as DeviceCapabilities from '@libs/DeviceCapabilities';
 import BaseHTMLEngineProvider from './BaseHTMLEngineProvider';
-import {HTMLEngineProviderProps} from './types';
+import type {HTMLEngineProviderProps} from './types';
 
-function HTMLEngineProvider({debug = false, children = null}: HTMLEngineProviderProps) {
+function HTMLEngineProvider({children = null}: HTMLEngineProviderProps) {
```
Comment (s77rt) [inline src/components/HTMLEngineProvider/index.tsx:7]:
```suggestion
function HTMLEngineProvider({children}: HTMLEngineProviderProps) {
```

No default value is needed since the prop is required

File: src/components/HTMLEngineProvider/htmlEngineUtils.ts [hunk 10]
```diff
@@ -47,15 +47,15 @@ function isChildOfNode(tnode: TNode, predicate: Predicate): boolean {
  * Finding node with name 'comment' flags that we are rendering a comment.
  */
 function isChildOfComment(tnode: TNode): boolean {
-    return isChildOfNode(tnode, (node) => isCommentTag(node.domNode?.name));
+    return isChildOfNode(tnode, (node) => node.domNode?.name !== undefined && isCommentTag(node.domNode?.name));
```
Comment (s77rt) [inline src/components/HTMLEngineProvider/htmlEngineUtils.ts:50]:
```suggestion
    return isChildOfNode(tnode, (node) => node.domNode?.name !== undefined && isCommentTag(node.domNode.name));
```

File: src/components/HTMLEngineProvider/htmlEngineUtils.ts [hunk 11]
```diff
@@ -47,15 +47,15 @@ function isChildOfNode(tnode: TNode, predicate: Predicate): boolean {
...
 }
 
 /**
  * Check if there is an ancestor node with the name 'h1'.
  * Finding a node with the name 'h1' flags that we are rendering inside an h1 element.
  */
 function isChildOfH1(tnode: TNode): boolean {
-    return isChildOfNode(tnode, (node) => node.domNode?.name.toLowerCase() === 'h1');
+    return isChildOfNode(tnode, (node) => node.domNode?.name !== null && node.domNode?.name.toLowerCase() === 'h1');
```
Comment (s77rt) [inline src/components/HTMLEngineProvider/htmlEngineUtils.ts:58]:
```suggestion
    return isChildOfNode(tnode, (node) => node.domNode?.name !== undefined && node.domNode.name.toLowerCase() === 'h1');
```

The `name` is never null, it's either a string or undefined so the condition will always be true and it will cause the app to crash

**Review by s77rt — CHANGES_REQUESTED** [review]

**Comment by ruben-rebelo** [comment]
Hi @s77rt, thank you so much again.
I've applied this changes!

**Comment by s77rt** [comment]
## Reviewer Checklist

- [X] I have verified the author checklist is complete (all boxes are checked off).
- [X] I verified the correct issue is linked in the `### Fixed Issues` section above
- [X] I verified testing steps are clear and they cover the changes made in this PR
    - [X] I verified the steps for local testing are in the `Tests` section
    - [X] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [X] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [X] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I verified tests pass on **all platforms** & I tested again on:
    - [X] Android: Native
    - [X] Android: mWeb Chrome
    - [X] iOS: Native
    - [X] iOS: mWeb Safari
    - [X] MacOS: Chrome / Safari
    - [X] MacOS: Desktop
- [X] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [X] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [X] If the PR modifies the form input styles:
    - [X] I verified that all the inputs inside a form are aligned with each other.
    - [X] I added `Design` label so the design team can review the changes.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>


android

</details>

<details>
<summary>Android: mWeb Chrome</summary>


mweb-chrome

</details>

<details>
<summary>iOS: Native</summary>


ios

</details>

<details>
<summary>iOS: mWeb Safari</summary>


mweb-safari

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>


web

</details>

<details>
<summary>MacOS: Desktop</summary>


desktop

</details>

**Comment by s77rt** [comment]
@ruben-rebelo Thank you!

**Review by s77rt — APPROVED** [review]

**Comment by ruben-rebelo** [comment]
Hello @mountiny, It seems there was an error with the melvin-bot and it didn't assign any engineer from Expensify.
Can you please give us an hand here?
Thank you!

**Comment by s77rt** [comment]
@nkuoch

**Review by nkuoch — APPROVED** [review]
